File: Simulink/ASM/parameter.m
%% PARAMETERS
%  ────────────────────────────────────────────────────────────────────────
%  Author:  SZ                  
%  Date:    08.10.2014                              
%  Version: V01R00
%  ────────────────────────────────────────────────────────────────────────
% ASM parameters
%
% Copyright 2014 Inspire AG, ETHZ. All rights reserved

%  ╒═════════╤═════════════╤═════════════╤═══════════════════════════════╕
%  │ VERSION │ DATE        │ AUTHOR      │ DESCRIPTION                   │
%  ╞═════════╪═════════════╪═════════════╪═══════════════════════════════╡
%  │ V01R00  │ 08.10.2014  │ SZ          │ Initial creation              │
%  ├─────────┼─────────────┼─────────────┼───────────────────────────────┤
%  │         │             │             │                               │
%  ╘═════════╧═════════════╧═════════════╧═══════════════════════════════╛

%% Motor
motor.J    = .00117;
motor.Rs   = .114;
motor.Rr   = .107;
motor.Ls   = .177e-3;
motor.Lr   = .267e-3;
motor.Lh   = 4.07e-3;
motor.p    = 2;
motor.k    = .66;        % Us/fs, where fs<fb
motor.Umax = 380;

%% Inputs
input.U  = 330;
input.fs = 500;
input.T  = 10e3/(14700/30*pi);
input.w0 = 14700/30*pi;


%% PWM
pwm.Umax = 400;
pwm.f    = 20e3;

File: ch.ethz.inspire.emod/src/ch/ethz/inspire/emod/model/parameters/Parameter.java
/***********************************
 * $Id$
 *
 * $URL$
 * $Author$
 * $Date$
 * $Rev$
 *
 * Copyright (c) 2011 by Inspire AG, ETHZ
 * All rights reserved
 *
 ***********************************/
package ch.ethz.inspire.emod.model.parameters;

/**
 * Implements a parameter with a unit and a describing comment
 * @author sizuest
 *
 * @param <T>
 */
public class Parameter<T> {
	private String name;
	private T value;
	private String comment;

	/**
	 * @param name
	 * @param value
	 */
	public Parameter(String name, T value) {
		this.name = name;
		this.value = value;
	}

	/**
	 * Get the value
	 * @return
	 */
	public T getValue() {
		return value;
	}

	/**
	 * Set the value
	 * @param value
	 */
	public void setValue(T value) {
		this.value = value;
	}

	/**
	 * Get the description
	 * @return
	 */
	public String getComment() {
		return this.comment;
	}

	/**
	 * Set the description
	 * @param comment
	 */
	public void setComment(String comment) {
		this.comment = comment;
	}

	/**
	 * Get the name
	 * @return
	 */
	public String getName() {
		return this.name;
	}

	/**
	 * Set the name
	 * @param name
	 */
	public void setName(String name) {
		this.name = name;
	}
}
